src/components/campaign/addCampaign/AddCampaign.js

```diff
@@ -1,53 +1,52 @@
 import { useState } from "react";
-import '../../styles/addCampaign.scss';
+import '../../../styles/addCampaign.scss';
 import { useDispatch, useSelector } from "react-redux";
-import { setCampaigns } from "../../features/campaigns";
+import { setCampaigns } from "../../../features/campaignSlice";
 
 const AddCampaign = () => {
     const dispatch = useDispatch();
     const campaigns = useSelector(state=>state.campaigns)
 
     const [ values, setValues ] = useState({});
     const [ errors, setErrors ] = useState({});
- 
 
-    const test= (e) => {
+    const handleSubmit= (e) => {
         e.preventDefault();
-        // const newCampaign = {
-        //     id:id,
-        //     name: name,
-        //     startDate: startDate,
-        //     endDate: endDate,
-        //     budget: budget
-        // }
-        let tempo = JSON.parse(JSON.stringify(campaigns.value))
-        // console.log(values)
-        tempo.push(values);
-        dispatch(setCampaigns(tempo))
+
+        const newCampaign = {...values}
+        if (!newCampaign.name || newCampaign.name === '') {
+            newCampaign.name = 'Unknown user'
+        };
+
+        let newCampArr = [...campaigns.value]
+        newCampArr.push(newCampaign);
+
+        dispatch(setCampaigns(newCampArr))
         setValues({});
+        e.target.reset();
     }
 
     const validate =(e, name, value) => {
         switch(name) {
-            case 'campaignId' :
+            case 'id' :
```

Comment:
`isNaN` is true for negative values :) sometimes even ID's comming from server can be negative, ok. 

But for budget, you should check isNaN, and `budget >= 0`